src/injected/web/gm-wrapper.js

```diff
@@ -305,15 +302,38 @@ function makeGlobalWrapper(local) {
       return true;
     },
   });
+  const mapWindow = desc => {
+    if (desc && (desc.value === global || desc.value === window)) {
+      desc = {
+        ...desc,
+        value: wrapper,
+      };
+    }
+    return desc;
+  };
+  for (const [name, desc] of unforgeables) {
+    defineProperty(local, name, mapWindow(desc));
+  }
+  // non-configurable properties in `global` should also be defined in `local` (target object)
+  // Reference: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Proxy/handler/getOwnPropertyDescriptor#Invariants
+  //
+  // > A property cannot be reported as non-configurable,
+  // > if it does not exists as an own property of the target object
+  // > or if it exists as a configurable own property of the target object.
+  for (const name of globals) {
+    if (!unforgeables.has(name)) {
+      const desc = describeProperty(global, name);
```

Comment:
This will be slow. I did that initially but it increases the startup time of page sandbox several times from 10-15ms to up to 50ms on my machine so it'll be slower for a lot of people.

We were checking the configurability on individual properties previously before my tweaks so you can just restore that (in `getOwnPropertyDescriptor()`).

Reply:
This has to be done if we are using *Proxy*. See the second section described above.

Reply:
We weren't enumerating the entire globals in 2.12.7 and Proxy worked correctly.

Reply:
I see the source of the bug is my new code:
```js
if (desc && desc.value === window) desc.value = wrapper;
```
Turns out I can't change the descriptor's value - it should be set on `local` initially in this loop:
```js
  for (const [name, desc] of unforgeables) {
    defineProperty(local, name, desc);
  }
```
Gonna poke around a little more...

Reply:
Ah I see, you are right.

Reply:
Here's a quick/dirty fix that illustrates a possible solution which I didn't test a lot though so see if you can come up with a better solution: https://puu.sh/Fomps/02f18e5873.txt